### ISSUE TITLE
performance gap with pytorch 0.4.1: mIoU 79.99

### ISSUE THREAD

**Issue by Wangzhuoying0716** [HRNet/HRNet-Semantic-Segmentation#67]
Sorry to bother you! I tried to implement the training process using the seg_hrnet_w48_train_512x1024_sgd_lr1e-2_wd5e-4_bs_12_epoch484.yaml. But it came out that the validation mIoU for single-scale and no-flip is 79.99, almost 1% lower than your result shown(80.9). I wonder if this is normal variation or I did something wrong ? 
Here is the training log:
https://pan.baidu.com/s/1utKUVuBEjDBtfgOk7A-5sQ  【passward: jckv】
Thank you!

**Comment by Hussainflr** [comment]
Hi, @Wangzhuoying0716  would mind sharing your training settings, 
I have been trying to train it, but it always gets stuck at this point 
![image](https://user-images.githubusercontent.com/49637191/68072389-176bce00-fdc0-11e9-8e2a-d9e0f3dfd8bb.png)


I don't know where I'm doing something wrong.

my training settings:
Pytorch 0.4.1
python 3.7
OpenCV 3.4.2.17
rest is according to requirements.txt

**Comment by Wangzhuoying0716** [comment]
@Hussainflr If there is no error message but just gets stuck, I guess it's because the memory is too full to keep on? You can check the memory usage when it is stuck to see whether it's almost full.
As for me,  I just used 4xTitan Xp GPU cards and the setting in  seg_hrnet_w48_train_512x1024_sgd_lr1e-2_wd5e-4_bs_12_epoch484.yaml.

**Comment by Wangzhuoying0716** [comment]
@Hussainflr I also used 
Pytorch 0.4.1 and python 3.7

**Comment by sunke123** [comment]
@Wangzhuoying0716 
Sorry for late reply.
I checked your log. When training on cityscapes, we use the "class balance" in the loss function.
You can change the config and set the class balance as True.

**Comment by whiteinblue** [comment]
Hi @sunke123, LOSS.CLASS_BALANCE defined in lib/config/default.py, but is there any place to use it? 
and i found, the loss: criterion = CrossEntropy(ignore_label=config.TRAIN.IGNORE_LABEL, 
 weight=train_dataset.class_weights), use class weighs, which doesn't matter with CLASS_BALANCE param

**Comment by sunke123** [comment]
@whiteinblue 
Class weights  are used to train the model on Cityscapes. If you don't want to use it, you can change it in the yaml.

**Comment by whiteinblue** [comment]
@sunke123 thank you for your reply， and i confuse, it's where that the param LOSS.CLASS_BALANCE used ?

**Comment by XuShoweR** [comment]
> Hi, @Wangzhuoying0716 would you mind sharing your training settings,
> I have been trying to train it, but it always gets stuck at this point
> ![image](https://user-images.githubusercontent.com/49637191/68072389-176bce00-fdc0-11e9-8e2a-d9e0f3dfd8bb.png)
> 
> I don't know where I'm doing something wrong.
> 
> my training settings:
> Pytorch 0.4.1
> python 3.7
> OpenCV 3.4.2.17
> rest is according to requirements.txt

Hello, i met same problem,  have you solved it?

**Comment by sunke123** [comment]
@whiteinblue Class_balance is used in the loss function.

**Comment by purse1996** [comment]
So where that the param LOSS.CLASS_BALANCE used?  I read the code, it seems that no matter what is CLASS_BALANCE param, the balance loss is used in the cityscape dataset.
